src/utils/CatalystClientBuilder.ts

```diff
@@ -35,114 +39,12 @@ export async function clientConnectedToCatalystIn(
 
 async function isServerUp(client: CatalystClient): Promise<boolean> {
   try {
-    await Promise.all([client.fetchContentStatus(), client.fetchLambdasStatus()])
-    return true
-  } catch {
-    return false
-  }
-}
-
-async function getApprovedListFromContract(network: 'mainnet' | 'ropsten'): Promise<string[]> {
-  const servers = network === 'mainnet' ? await getMainnetCatalysts() : await getRopstenCatalysts()
-  return servers.map(({ address }) => address)
-}
-
-/**
- * The idea here is to build an updated list of the catalysts approved by the DAO, without actually querying the DAO's contract
- * directly. This is because that query is both slow and expensive, so the idea is to use a list of known servers and ask them
- * for the updated list. The thing is this known server list might be outdated, so we need to take that into account. We will
- * take N (defined by REQUIRED_LISTS below) random servers from this known list, and ask them for the currently approved ones. We will then use the intersection of those
- * N lists as the updated list.
- */
-const REQUIRED_LISTS = 3
-export async function getUpdatedApprovedListWithoutQueryingContract({
-  preKnownServers,
-  origin,
-  requiredLists,
-  fetchApprovedCatalysts
-}: {
-  preKnownServers: { list: { address: string }[] } | { network: 'mainnet' | 'ropsten' }
-  origin: string
-  requiredLists?: number
-  fetchApprovedCatalysts?: (catalystUrl: string, origin: string) => Promise<string[] | undefined>
-}): Promise<string[] | undefined> {
-  // Set defaults if needed
-  const catalystListFetch =
-    fetchApprovedCatalysts ?? ((catalystUrl, origin) => fetchCatalystsApprovedByDAO(catalystUrl, origin))
-  const requiredAmountOfLists = requiredLists ?? REQUIRED_LISTS
-
-  // Get the list of known servers
-  const knownServers = 'list' in preKnownServers ? preKnownServers.list : CatalystsList[preKnownServers.network]
-
-  // If I don't know enough servers, then it doesn't make sense to continue
-  if (knownServers.length < requiredAmountOfLists) {
-    return undefined
-  }
-
-  // Shuffle the list
-  const shuffledPreKnownServers = shuffleArray(knownServers)
-
-  // Ask N of them for their list
-  const approvedServersList = await Promise.all(
-    shuffledPreKnownServers
-      .slice(0, requiredAmountOfLists + 3)
-      .map((server) => server.address)
-      .map((address) => catalystListFetch(address, origin))
-  )
+    const result = await client.fetchPeerHealth()
```

Comment:
Consider adding a timeout here. It should have been there in the first place, but it's a good time to add it